packages/features/calendar-cache/lib/datesForCache.ts

```diff
@@ -1,16 +1,66 @@
 /** Expand the start date to the beginning of the current month */
 export const getTimeMin = (timeMin?: string) => {
   const date = timeMin ? new Date(timeMin) : new Date();
-  date.setDate(1);
-  date.setHours(0, 0, 0, 0);
-  return date.toISOString();
+  // Set to UTC to avoid timezone issues
+  const result = new Date(Date.UTC(date.getUTCFullYear(), date.getUTCMonth(), 1));
+  result.setUTCHours(0, 0, 0, 0);
+  return result.toISOString();
 };
 
-/** Expand the end date to the end of the next month */
+/**
+ * Expand the end date to the start of the overnext month if date
+ * is between start of current month and end of next month,
+ * otherwise return start of overnext month from the passed date
+ * @example
+ * Today: March 15, 2024 ▼-------------▼ getTimeMax returns May 1st 00:00:00
+ *                   Mar      Apr      May      Jun
+ *                   ├────────┼────────┼────────┼────────┤
+ * Current Month     █████████|        |        |        |
+ * Next Month        |        █████████|        |        |
+ **/
 export function getTimeMax(timeMax?: string) {
-  const date = timeMax ? new Date(timeMax) : new Date();
-  date.setMonth(date.getMonth() + 1);
-  date.setDate(0);
-  date.setHours(0, 0, 0, 0);
-  return date.toISOString();
+  const now = new Date();
+  const currentMonth = now.getUTCMonth();
+  const currentYear = now.getUTCFullYear();
+
+  if (!timeMax) {
+    // If no date passed, return start of the overnext month from current date
+    const result = new Date(Date.UTC(currentYear, currentMonth + 2, 1));
+    result.setUTCHours(0, 0, 0, 0);
+    return result.toISOString();
+  }
+
+  const date = new Date(timeMax);
+  const dateMonth = date.getUTCMonth();
+  const dateYear = date.getUTCFullYear();
+
+  // If the date is within current month or next month, return start of overnext month
+  // Otherwise, return start of overnext month from the date
+  let targetYear = dateYear;
+  let targetMonth = dateMonth;
+
+  // Check if date is within current month or next month
+  const isWithinCurrentOrNextMonth =
+    (dateYear === currentYear && dateMonth <= currentMonth + 1) ||
+    (dateYear === currentYear + 1 && currentMonth === 11 && dateMonth === 0);
+
+  if (isWithinCurrentOrNextMonth) {
+    // For dates within current month or next month, return start of overnext month
+    targetMonth = dateMonth + 2;
+    if (targetMonth > 11) {
+      targetMonth = 0;
+      targetYear++;
+    }
+  } else {
+    // For dates beyond overnext month, return start of overnext month from the date
+    targetMonth = dateMonth + 2;
+    if (targetMonth > 11) {
+      targetMonth = 0;
```

Comment:
This doesn't look 100% correct to me in a certain situation:

targetMonth = 11 + 2;
if (13 > 11) {
  targetMonth = 0 (= 12)
  targetYear = currentYear + 1;
}

Should target month in this situation be January? Or should it be February and this code has an error?